### ISSUE TITLE
Add a "Request starting …" log to RequestLoggingMiddleware

### ISSUE THREAD

**Issue by JSkimming** [serilog/serilog-aspnetcore#171]
#### TL;DR

Would you consider adding a log at the beginning of the request, similar to the original logged from `Microsoft.AspNetCore.Hosting.Diagnostics`. I'd be happy to submit a PR, but thought it worth checking if it will be happily received before cutting code 😃.

#### Background

I followed the [current guidelines to enable Request logging](https://github.com/serilog/serilog-aspnetcore/blob/v3.2.0/README.md#request-logging "README Request logging section"). I noticed I had suppressed two logs from `Microsoft.AspNetCore.Hosting.Diagnostics` which were replaced by a single log from [RequestLoggingMiddleware](https://github.com/serilog/serilog-aspnetcore/blob/v3.2.0/src/Serilog.AspNetCore/AspNetCore/RequestLoggingMiddleware.cs#L74-L100 "RequestLoggingMiddleware.cs").

At the time I thought nothing about it until we hit an issue in production where a request never completed, and unsurprisingly, we never got to see the log from `RequestLoggingMiddleware`.

Unfortunately, this meant we were missing the key piece of information to aid debugging, that being a query string parameter. The full path and query is logged by `Microsoft.AspNetCore.Hosting.Diagnostics` (and `RequestLoggingMiddleware`) but is not part of the diagnostic context and as a result no part of any subsequent logs (the URL path is part of the diagnostic context, but not the query string).

So my reasoning is if `RequestLoggingMiddleware` replaces the equivalent logs of  `Microsoft.AspNetCore.Hosting.Diagnostics` maybe it should also include a log at the start of a request as well as the end and include all the same information.

**Comment by nblumhardt** [comment]
Hi! Thanks for the note.

The `RequestLoggingMiddleware` can, but doesn't have to, replace the built-in logs. It sounds like your scenario might be handled cleanly by adding `MinimumLevel.Override("Microsoft.AspNetCore.Hosting.Diagnostics", LogEventLevel.Information)` to you Serilog config - does that help?

At the moment I think the goal of `UseSerilogRequestLogging()` is to condense things to the single event - I can see some value in multiple events per request, but I think that will open up a lot more scope that's better addressed through selectively enabling the built-in logs.

**Comment by ahdung** [comment]
Hows going on now? I really need request start log.
